src/simple_commander/utils/line_intersection.py

```diff
@@ -3,36 +3,49 @@
 import math
 
 
-def line_intersection(line1, line2):
+def line_intersection(line1, line2, width1, width2):
```

Comment:
If I found the main sense of changes, You try to find not only one line intersection, in code you detect intersection of units from different sides of the object.
If it's true, please rename the name of function and comment. Original code was working only for line, that's why name is "line_intersection"